src/kvstore/kvstore_dist.h

```diff
@@ -237,7 +237,8 @@ class KVStoreDist : public KVStoreLocal {
       // TODO(haibin) refactor this for loop
       for (size_t i = 0; i < num_vals; i++) {
         auto &row_id = target_val_rowids[i].second;
-        NDArray indices = row_id.Copy(pinned_ctx_);
+        NDArray indices(row_id.shape(), pinned_ctx_, false, mshadow::kInt64);
```

Comment:
?

Reply:
This makes sure rsp_pull(val, rowid) can accept rowid with dtype other than int64.